Source/NetworkReachabilityManager.swift

```diff
@@ -0,0 +1,208 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+import SystemConfiguration
+
+/// Notification posted when network reachability status changes. The notification `object` contains the update
+/// network reachability status as an `NSNumber` which will need to be converted.
+public let NetworkReachabilityStatusDidChangeNotification = "com.alamofire.network.reachability.status.did.change"
+
+/**
+    Defines the various states of network reachability.
+
+    - Unknown:         It is unknown whether the network is reachable.
+    - NotReachable:    The network is not reachable.
+    - ReachableOnWWAN: The network is reachable over the WWAN connection.
+    - ReachableOnWiFi: The network is reachable over the WiFi connection.
+*/
+public enum NetworkReachabilityStatus: Int {
+    case Unknown         = -1
+    case NotReachable    = 0
+    case ReachableOnWWAN = 1
+    case ReachableOnWiFi = 2
+}
+
+/**
+    The `NetworkReachabilityManager` class listens for reachability changes of hosts and addresses for both WWAN and
+    WiFi network interfaces.
+
+    Reachability can be used to determine background information about why a network operation failed, or to retry
+    network requests when a connection is established. It should not be used to prevent a user from initiating a network
+    request, as it's possible that an initial request may be required to establish reachability.
+*/
+public class NetworkReachabilityManager {
+    /// A closure executed when the network reachability status changes. The closure takes a single argument: the 
+    /// network reachability status.
+    public typealias Listener = NetworkReachabilityStatus -> Void
+
+    // MARK: - Properties
+
+    /// Whether the network is currently reachable.
+    public var isReachable: Bool { return isReachableOnWWAN || isReachableOnWiFi }
+
+    /// Whether the network is currently reachable over the WWAN interface.
+    public var isReachableOnWWAN: Bool { return networkReachabilityStatus == .ReachableOnWWAN }
```

Comment:
Yes, much of this API will have to be platform specific, perhaps even the definition of `NetworkReachabilityStatus`, since, for OS X, it would be more appropriately defined as:
```swift
public enum NetworkReachabilityStatus: Int {
    case Unknown
    case NotReachable
    case Reachable
}
```
Since OS X doesn't distinguish reachability by interface.